wordpress-mobile/WordPress-iOS
# Add Button to Milestone Notifications detail view

This PR adds the ability for notifications to show a button in their detail view. Notification text blocks with a `meta` value of `"button": true` will be assigned the `button` kind, and rendered using the new `NoteBlockButtonTableViewCell`.

<img src="https://user-images.githubusercontent.com/4780/108764100-0feaae00-754a-11eb-8747-f6b3f7f644cd.PNG" width=300>

**Note**

The button currently takes you to the notifications day view for the specified site, but our system for routing those deep links currently doesn't respect the date range passed to it. We'll tweak that in a future PR.

**To test**

- You'll need to have the backend diff for Milestone Notifications applied to your sandbox and your simulator or device set up to use this for API calls.
- Ensure you have a Milestone Notification present in your notifications list.
- Enable the milestone notifications feature flag.
- Build and run and view the notification. Check that the **Take a deep dive button** is present.
- Check other notifications, and ensure that the button doesn't show up elsewhere.
- Ensure tests pass.
- Ensure that the button doesn't show up if the feature flag is disabled.

PR submission checklist:
